internal/libyaml/yamlh.go

```diff
@@ -0,0 +1,804 @@
+	case BLOCK_END_TOKEN:
+		return "BLOCK_END_TOKEN"
+	case FLOW_SEQUENCE_START_TOKEN:
+		return "FLOW_SEQUENCE_START_TOKEN"
+	case FLOW_SEQUENCE_END_TOKEN:
+		return "FLOW_SEQUENCE_END_TOKEN"
+	case FLOW_MAPPING_START_TOKEN:
+		return "FLOW_MAPPING_START_TOKEN"
+	case FLOW_MAPPING_END_TOKEN:
+		return "FLOW_MAPPING_END_TOKEN"
+	case BLOCK_ENTRY_TOKEN:
+		return "BLOCK_ENTRY_TOKEN"
+	case FLOW_ENTRY_TOKEN:
+		return "FLOW_ENTRY_TOKEN"
+	case KEY_TOKEN:
+		return "KEY_TOKEN"
+	case VALUE_TOKEN:
+		return "VALUE_TOKEN"
+	case ALIAS_TOKEN:
+		return "ALIAS_TOKEN"
+	case ANCHOR_TOKEN:
+		return "ANCHOR_TOKEN"
+	case TAG_TOKEN:
+		return "TAG_TOKEN"
+	case SCALAR_TOKEN:
+		return "SCALAR_TOKEN"
+	}
+	return "<unknown token>"
+}
+
+// The token structure.
+type Token struct {
+	// The token type.
+	typ TokenType
+
+	// The start/end of the token.
+	start_mark, end_mark Mark
+
+	// The stream encoding (for STREAM_START_TOKEN).
+	encoding Encoding
+
+	// The alias/anchor/scalar value or tag/tag directive handle
+	// (for ALIAS_TOKEN, ANCHOR_TOKEN, SCALAR_TOKEN, TAG_TOKEN, TAG_DIRECTIVE_TOKEN).
+	value []byte
+
+	// The tag suffix (for TAG_TOKEN).
+	suffix []byte
+
+	// The tag directive prefix (for TAG_DIRECTIVE_TOKEN).
+	prefix []byte
+
+	// The scalar style (for SCALAR_TOKEN).
+	style ScalarStyle
+
+	// The version directive major/minor (for VERSION_DIRECTIVE_TOKEN).
+	major, minor int8
+}
+
+// Events
+
+type EventType int8
+
+// Event types.
+const (
+	// An empty event.
+	NO_EVENT EventType = iota
+
+	STREAM_START_EVENT   // A STREAM-START event.
+	STREAM_END_EVENT     // A STREAM-END event.
+	DOCUMENT_START_EVENT // A DOCUMENT-START event.
+	DOCUMENT_END_EVENT   // A DOCUMENT-END event.
+	ALIAS_EVENT          // An ALIAS event.
+	SCALAR_EVENT         // A SCALAR event.
+	SEQUENCE_START_EVENT // A SEQUENCE-START event.
+	SEQUENCE_END_EVENT   // A SEQUENCE-END event.
+	MAPPING_START_EVENT  // A MAPPING-START event.
+	MAPPING_END_EVENT    // A MAPPING-END event.
+	TAIL_COMMENT_EVENT
+)
+
+var eventStrings = []string{
+	NO_EVENT:             "none",
+	STREAM_START_EVENT:   "stream start",
+	STREAM_END_EVENT:     "stream end",
+	DOCUMENT_START_EVENT: "document start",
+	DOCUMENT_END_EVENT:   "document end",
+	ALIAS_EVENT:          "alias",
+	SCALAR_EVENT:         "scalar",
+	SEQUENCE_START_EVENT: "sequence start",
+	SEQUENCE_END_EVENT:   "sequence end",
+	MAPPING_START_EVENT:  "mapping start",
+	MAPPING_END_EVENT:    "mapping end",
+	TAIL_COMMENT_EVENT:   "tail comment",
+}
+
+func (e EventType) String() string {
+	if e < 0 || int(e) >= len(eventStrings) {
+		return fmt.Sprintf("unknown event %d", e)
+	}
+	return eventStrings[e]
+}
+
+// The event structure.
+type Event struct {
+	// The event type.
+	Type EventType
+
+	// The start and end of the event.
+	StartMark, EndMark Mark
+
+	// The document encoding (for STREAM_START_EVENT).
+	encoding Encoding
+
+	// The version directive (for DOCUMENT_START_EVENT).
+	version_directive *VersionDirective
```

Comment:
@ccoVeille the casing of this field is totally different from the casing of eg. the StartMark field.